### PR TITLE
Combine url and preflight frames in iroh transport

### LINKED ISSUE #399: Refactor iroh handshake to combine peer URL and preflight
Currently the peer URL is sent first, followed by the preflight, when establishing a new connection. This can be simplified by concatenating peer URL and preflight or otherwise combine them with a protobuf encoded message, resulting in just one handshake message before sending some actual payload message.

### PR DESCRIPTION
**Pull request by jost-s**
resolves #399 and #384

## Summary by CodeRabbit

* **Refactor**
  * Framing redesigned to explicit Preflight and Data frames; preflight exchanged first and data processed only after handshake.
* **Bug Fixes**
  * Stronger frame validation, URL parsing, and clearer error/teardown paths for malformed frames and preflight failures.
* **New Features**
  * Public frame encode/decode surface and unified send API; per-connection preflight state accessors; transport config option to allow plaintext relay in tests.
* **Tests**
  * Updated and expanded unit/integration tests covering Preflight/Data flows and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: crates/transport_iroh/src/frame.rs [hunk 1]
```diff
@@ -44,61 +119,64 @@ pub(super) fn decode_frame(data: Vec<u8>) -> K2Result<(FrameType, Bytes)> {
...
+            let url = Url::from_str(
+                std::str::from_utf8(&payload[4..4 + url_len]).map_err(
+                    |err| K2Error::other_src("invalid peer url", err),
+                )?,
+            )?;
+            // The preflight takes up the rest of the payload,
+            // after the URL length and the URL.
+            let preflight_bytes =
+                Bytes::copy_from_slice(&payload[4 + url_len..]);
+            Frame::Preflight((url, preflight_bytes))
```
Comment (mattyg) [inline crates/transport_iroh/src/frame.rs]: Should we error on empty preflight bytes?

**Review by ThetaSinner — APPROVED** [review]
